plugin/src/main/groovy/com/btkelly/gnag/api/GitHubApi.java

```diff
@@ -53,41 +57,78 @@ public GitHubApi(final GitHubExtension gitHubExtension) {
         Retrofit retrofit = new Retrofit.Builder()
                 .baseUrl(baseUrl)
                 .client(okHttpClient)
+                .addConverterFactory(DiffParserConverterFactory.create())
                 .addConverterFactory(GsonConverterFactory.create())
                 .build();
 
         gitHubApiClient = retrofit.create(GitHubApiClient.class);
     }
 
-    public Status postGitHubComment(String comment) {
+    public void postGitHubPRCommentAsync(final String comment) {
+        gitHubApiClient.postPRComment(new GitHubPRComment(comment), gitHubExtension.getIssueNumber())
+                .enqueue(new DefaultCallback<>());
+    }
+
+    public void postUpdatedGitHubStatusAsync(final GitHubStatusType gitHubStatusType, final String prSha) {
+        gitHubApiClient.postUpdatedStatus(new GitHubStatus(gitHubStatusType), prSha)
+                .enqueue(new DefaultCallback<>());
+    }
 
+    @Nullable
+    public GitHubPRDetails getPRDetailsSync() {
         try {
-            Response<GitHubComment> gitHubCommentResponse = gitHubApiClient.postComment(new GitHubComment(comment), gitHubExtension.getIssueNumber()).execute();
-            return gitHubCommentResponse.isSuccessful() ? Status.OK : Status.FAIL;
-        } catch (IOException ignored) {
-            return Status.FAIL;
+            Response<GitHubPRDetails> gitHubPRDetailsResponse = gitHubApiClient.getPRDetails(gitHubExtension.getIssueNumber()).execute();
+            return gitHubPRDetailsResponse.body();
+        } catch (final Exception e) {
+            e.printStackTrace();
+            return null;
         }
     }
 
-    public Status postUpdatedGitHubStatus(GitHubStatusType gitHubStatusType, String sha) {
-
+    @NotNull
+    public List<Diff> getPRDiffsSync() {
         try {
-            Response<GitHubStatus> gitHubStatusResponse = gitHubApiClient.postUpdatedStatus(new GitHubStatus(gitHubStatusType), sha).execute();
-            return gitHubStatusResponse.isSuccessful() ? Status.OK : Status.FAIL;
-        } catch (IOException ignored) {
-            return Status.FAIL;
+            final Response<List<Diff>> gitHubPRDiffsResponse
+                    = gitHubApiClient.getPRDiffs(gitHubExtension.getIssueNumber()).execute();
+            
+            return gitHubPRDiffsResponse.body();
+        } catch (final Exception e) {
+            e.printStackTrace();
+            return new ArrayList<>();
         }
     }
 
-    public GitHubPullRequest getPullRequestDetails() {
+    public void postGitHubInlineCommentSync(
```

Comment:
These are currently posted synchronously and sequentially. When I attempted to use OkHttp to `enqueue` all of these as asynchronous requests, only one comment would ever post. I'm not sure if this was because I was misunderstanding how OkHttp manages threads, or because the GitHub API (secretly) requires comments to be posted 'atomically'. By inspecting Retrofit output, I was able to see that all inline comments were being submitted by gnag, but I was unable to receive responses to inspect (I wonder if this was because the `Call` is defined with a `Void` type? Could try with `Object`?